ddtrace/appsec/_utils.py

```diff
@@ -1,29 +1,25 @@
 import os
-from typing import TYPE_CHECKING
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Union
+from urllib import parse
 
 from ddtrace.appsec import _asm_request_context
 from ddtrace.appsec._constants import API_SECURITY
 from ddtrace.constants import APPSEC_ENV
-from ddtrace.internal.compat import parse
 from ddtrace.internal.logger import get_logger
 from ddtrace.internal.utils.http import _get_blocked_template  # noqa
 from ddtrace.settings import _config as config
 
 
-if TYPE_CHECKING:  # pragma: no cover
-    from typing import Any
-
-    from ddtrace.internal.compat import text_type as unicode
-
-
 log = get_logger(__name__)
 
 
-def parse_form_params(body):
-    # type: (unicode) -> dict[unicode, unicode|list[unicode]]
+def parse_form_params(body: str) -> Dict[str, Union[str, List[str]]]:
     """Return a dict of form data after HTTP form parsing"""
     body_params = body.replace("+", " ")
-    req_body = dict()  # type: dict[unicode, unicode|list[unicode]]
+    req_body: Dict[str, Union[str, list[str]]] = dict()
```

Comment:
```suggestion
    req_body: Dict[str, Union[str, List[str]]] = dict()
```
to ensure mypy checks the parameter for Python 3.7 and 3.8

Reply:
Thank you for catching that!